test/__init__.py

```diff
@@ -69,6 +69,7 @@
 import pymongo
 import pymongo.errors
 from bson.son import SON
+from pymongo import IndexModel  # noqa
```

Comment:
Needed why?

Reply:
`self.assertRepr` test fails without it.

Reply:
I haven't check yet but probably the calls to `self.assertRepr` from `test_connection…` will fail to without importing whatever type `repr` tries to use. So, we don't duplicate that code but we do have to import types being tested.

Reply:
And… too many imports… duplicating `assertRepr` code for now.

Reply:
Right... `eval()` needs it. If `assertRepr` lived in the base class, I think you could put the "noqa" imports inside the method. I don't have a strong opinion about whether that's cleaner than duplicating the method.